### PR TITLE
Add go caching in CI

### PR DESCRIPTION
**Pull request by lucacome**
Shaving off a couple of minutes in every step with caching for golang.

### CODE REVIEW

**Comment by soneillf5** [comment]
how have you tested this?

**Comment by lucacome** [comment]
> how have you tested this?

if you look at the logs in the old jobs you can see that go is downloading all the dependencies to run the tests or check for CRDs changes. With this change when there's a cache hit it won't have to download them. You can also compare the times for the tests/build form old runs with this one.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -50,6 +50,15 @@ jobs:
...
         uses: actions/checkout@v2
+      - name: Cache Go controller tools
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-tools-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-tools-
```
Comment (pleshakov) [inline .github/workflows/ci.yml:61]: what is the reason for configuring `restore-keys`?

Comment (lucacome) [reply .github/workflows/ci.yml:61]: it's used to search for partial matches if there's a cache miss for the key.

I just used the example from here btw https://github.com/actions/cache/blob/main/examples.md#go---modules

**Review by pleshakov — APPROVED** [review]

**Review by ciarams87 — APPROVED** [review]

**Pull request merged**